tests/suites/test_suite_aes.function

```diff
@@ -151,6 +151,157 @@ exit:
+    unhexify( iv_str, hex_iv_string );
+    data_len = unhexify( src_str, hex_src_string );
+
+    mbedtls_aes_xts_setkey_enc( &ctx, key_str, key_len * 8 );
+
+    TEST_ASSERT( mbedtls_aes_crypt_xex( &ctx, MBEDTLS_AES_ENCRYPT, data_len, iv_str, src_str, output ) == 0);
+    hexify( dst_str, output, data_len );
+    TEST_ASSERT( strcmp( (char *) dst_str, hex_dst_string ) == 0 );
+
+exit:
+    mbedtls_aes_xts_free( &ctx );
+}
+/* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_CIPHER_MODE_XTS */
+void aes_decrypt_xex( char *hex_key_string, char *hex_iv_string,
+                      char *hex_src_string, char *hex_dst_string )
+{
+    unsigned char key_str[100] = { 0, };
+    unsigned char iv_str[100]  = { 0, };
+    unsigned char src_str[100] = { 0, };
+    unsigned char dst_str[100] = { 0, };
+    unsigned char output[100]  = { 0, };
+    mbedtls_aes_xts_context ctx;
+    int key_len, data_len;
+
+    mbedtls_aes_xts_init( &ctx );
+
+    key_len = unhexify( key_str, hex_key_string );
+    unhexify( iv_str, hex_iv_string );
+    data_len = unhexify( src_str, hex_src_string );
+
+    mbedtls_aes_xts_setkey_dec( &ctx, key_str, key_len * 8 );
+
+    TEST_ASSERT( mbedtls_aes_crypt_xex( &ctx, MBEDTLS_AES_DECRYPT, data_len, iv_str, src_str, output ) == 0 );
+
+    hexify( dst_str, output, data_len );
+    TEST_ASSERT( strcmp( (char *) dst_str, hex_dst_string ) == 0 );
+
+exit:
+    mbedtls_aes_xts_free( &ctx );
+}
+/* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_CIPHER_MODE_XTS */
+void aes_encrypt_xts( char *hex_key_string, char *hex_data_unit_string,
+                      char *hex_src_string, char *hex_dst_string )
+{
+    enum { AES_BLOCK_SIZE = 16 };
+    unsigned char *key;
+    unsigned char *data_unit;
+    unsigned char *src;
+    unsigned char *dst;
+    unsigned char *output;
+    mbedtls_aes_xts_context ctx;
+    size_t key_len, src_len, dst_len, data_unit_len;
+
+    mbedtls_aes_xts_init( &ctx );
+
+    data_unit = unhexify_alloc( hex_data_unit_string, &data_unit_len );
+    TEST_ASSERT( data_unit_len == AES_BLOCK_SIZE );
+
+    key = unhexify_alloc( hex_key_string, &key_len );
+    TEST_ASSERT( key_len % 2 == 0 );
+
+    src = unhexify_alloc( hex_src_string, &src_len );
+    dst = unhexify_alloc( hex_dst_string, &dst_len );
+    TEST_ASSERT( src_len == dst_len );
+
+    output = zero_alloc(dst_len);
+
+    mbedtls_aes_xts_setkey_enc( &ctx, key, key_len * 8 );
+    TEST_ASSERT( mbedtls_aes_crypt_xts( &ctx, MBEDTLS_AES_ENCRYPT, src_len, data_unit, src, output ) == 0 ); // XXX Code Review: must this be one line?
+
+    TEST_ASSERT( memcmp( output, dst, dst_len ) == 0 );
+
+exit:
+    mbedtls_aes_xts_free( &ctx );
+}
+/* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_CIPHER_MODE_XTS */
+void aes_decrypt_xts( char *hex_key_string, char *hex_data_unit_string,
+                      char *hex_dst_string, char *hex_src_string )
+{
+    /* XXX Code Review: I don't really like all this duplicated code between
```

Comment:
I too like the idea of unifying them.

Reply:
Yeah, but I believe we should optimize for clarity when a test fails over the maintainability of the test. Adding a shared helper function is great for the maintainability of this code, but if that makes it harder to see where an error is coming from when a test fails, I'd rather pay the maintenance cost: it'd be worth it.

Would you agree? If so, I can delete this junk.

Reply:
When a test fails we know which test case failed and the printed info is the same whether we are using a helper function or not. If we don't know what is going on based on what is printed out, then we need to get our hands dirty anyway. I don't know how much reducing call depth helps when we are inserting printf's in the code or running a debugger anyway.

In summary I am still leaning towards unifying them, but don't consider this as a blocker.

Reply:
I'll leave them separate for now. I've removed my XXX comment requesting feedback. I think getting a line number from an assertion failing in a helper function is not so great.